adminSiteServer/apiRouter.ts

```diff
@@ -1962,7 +1962,7 @@ apiRouter.put("/datasets/:datasetId", async (req: Request, res: Response) => {
             commitEmail: res.locals.user.email,
         })
     } catch (err) {
-        logErrorAndMaybeSendToSlack(err)
+        logErrorAndMaybeSendToBugsnag(err)
```

Comment:
Looking at [the docs](https://docs.bugsnag.com/platforms/javascript/express/#reporting-handled-errors) it seems to me that we should include the request whenever possible to get the added context. I think to have the full flexbility we should add the request as a second parameter and in logErrorAndMaybeSendToBugsnag call `req.bugsnag.notify()`

Reply:
Make sense, thanks!

For reference, routes used for testing in app.tsx:
```js
        // routes to generate errors for testing
        app.get("/error", (req, res) => {
            throw new Error("This is a test error")
        })

        app.get("/log-error-with-req", (req, res) => {
            const error = new Error("This is a test error with req")
            logErrorAndMaybeSendToBugsnag(error, req)
            res.send("logged error")
        })

        app.get("/log-error-without-req", (req, res) => {
            const error = new Error("This is a test error without req")
            logErrorAndMaybeSendToBugsnag(error, req)
            res.send("logged error")
        })
```

Reply:
Logging error with `req`:

<img width="431" alt="Screenshot 2023-05-26 at 10 45 38" src="https://github.com/owid/owid-grapher/assets/13406362/f0f21efd-86e5-420b-a3f8-ce8f9b5e2061">